### PR TITLE
Do not leak java.lang.ProcessEnvironment.

### PR DESCRIPTION
**Pull request by radcortez**

### CODE REVIEW

**Review by gsmet — APPROVED** [review]
As mentioned by @gastaldi you could use `putAll` to get a better default sizing but other than that, looks good.

But I would add a comment for future you.

**Review by gastaldi — DISMISSED** [review]
image

**Comment by gastaldi** [comment]
You could maybe also use `Collections.unmodifiableMap` on the newly created `HashMap` to comply with the semantics of `java.lang.System#getenv()`

File: implementation/src/main/java/io/smallrye/config/EnvConfigSource.java [hunk 1]
```diff
@@ -110,12 +110,13 @@ private static String replaceNonAlphanumericByUnderscores(String name) {
         return sb.toString();
     }
 
+    /**
+     * A new Map with the contents of System.getEnv. Im the Windows implementation, the Map is an extension of
```
Comment (gastaldi) [inline implementation/src/main/java/io/smallrye/config/EnvConfigSource.java:114]:
```suggestion
     * A new Map with the contents of System.getEnv. In the Windows implementation, the Map is an extension of
```

File: implementation/src/main/java/io/smallrye/config/EnvConfigSource.java [hunk 2]
```diff
@@ -110,12 +110,13 @@ private static String replaceNonAlphanumericByUnderscores(String name) {
...
+     * ProcessEnvironment. This causes issues with Graal and native mode, since ProcessEnvironment should not be
+     * instanciated in the heaop.
```
Comment (gastaldi) [inline implementation/src/main/java/io/smallrye/config/EnvConfigSource.java:116]:
```suggestion
     * instanciated in the heap.
```

**Pull request merged**